autogpt_platform/backend/backend/blocks/example/triggers.py

```diff
@@ -0,0 +1,65 @@
+import logging
+
+from backend.data.block import (
+    Block,
+    BlockCategory,
+    BlockManualWebhookConfig,
+    BlockOutput,
+    BlockSchema,
+)
+from backend.data.model import SchemaField
+from backend.integrations.webhooks.example import ExampleWebhookEventType
+
+logger = logging.getLogger(__name__)
+
+
+class ExampleTriggerBlock(Block):
+    """
+    A trigger block that is activated by an external webhook event.
+
+    Unlike standard blocks that are manually executed, trigger blocks are automatically
+    activated when a webhook event is received from the specified provider.
+    """
+
+    class Input(BlockSchema):
+        # The payload field is hidden because it's automatically populated by the webhook
+        # system rather than being manually entered by the user
+        payload: dict = SchemaField(hidden=True)
+
+    class Output(BlockSchema):
+        event_data: dict = SchemaField(
```

Comment:
Does this have to be a single event_data and be parsed manually later by another block?
The best practice is to avoid returning complex object as much as possible.

Reply:
@Pwuts can you help with best practice here please